[ews-build.webkit.org] Convert AnalyzeAPITestsResults to new-style buildstep

JonWBedard opened the pull request:
#### ca0fcb99825e06ae225b212780785c51653b70eb
<pre>
[ews-build.webkit.org] Convert AnalyzeAPITestsResults to new-style buildstep
<a href="https://bugs.webkit.org/show_bug.cgi?id=251230">https://bugs.webkit.org/show_bug.cgi?id=251230</a>
rdar://104718250

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(AnalyzeAPITestsResults):
(AnalyzeAPITestsResults.run): Renamed from start.
(AnalyzeAPITestsResults.analyzeResults): Make entire function asynchronous, since it calls
asynchronous functions.
(AnalyzeAPITestsResults.getTestsResults): Correctly yield log additions.
(AnalyzeAPITestsResults.start): Renamed to run.

Canonical link: <a href="https://commits.webkit.org/259526@main">https://commits.webkit.org/259526@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ba478706ed1346174fc18b95f737bcc28e9588be

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104942 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/14022 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37839 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/114208 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/174403 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108847 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/15156 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4947 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97266 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/113235 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110702 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/11712 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94722 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/39229 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/93583 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/26343 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80894 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7366 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27702 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/7459 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/4289 "Found 1 new test failure: fast/forms/fieldset/fieldset-elements.html (failure)") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/103744 "Passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/13516 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/47254 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/9250 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3509 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

aj062 reviewed (approved):
r+ assuming this is tested.

JonWBedard committed:
[ews-build.webkit.org] Convert AnalyzeAPITestsResults to new-style buildstep
https://bugs.webkit.org/show_bug.cgi?id=251230
rdar://104718250

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(AnalyzeAPITestsResults):
(AnalyzeAPITestsResults.run): Renamed from start.
(AnalyzeAPITestsResults.analyzeResults): Make entire function asynchronous, since it calls
asynchronous functions.
(AnalyzeAPITestsResults.getTestsResults): Correctly yield log additions.
(AnalyzeAPITestsResults.start): Renamed to run.

Canonical link: https://commits.webkit.org/259526@main
